### PR TITLE
Docs: clarify Tensor.is_contiguous definition for expanded dimensions

### LINKED ISSUE #190038: `Tensor.is_contiguous` is false for expanded tensors
### 📚 The doc issue

I'm looking at https://docs.pytorch.org/docs/2.13/generated/torch.Tensor.is_contiguous.html which simply states
> Returns True if self tensor is contiguous in memory in the order specified by memory format.

However this "contiguous" is possibly ambiguous.
Background: I ran into an accuracy issue where one codepath was using an expanded tensor while the other left it unexpanded leading to different results in a matmul.
That boiled down to how "contiguous" is defined:

```
import torch
x=torch.tensor(1)
assert x.is_contiguous()
x2=x.expand((2,2))
assert not x2.is_contiguous()
assert x2.stride() == (0,0)
```

I.e. a tensor with 1 element in memory is not considered "contiguous". This matters in AoT-Autograd, e.g. in [`coerce_to_expected_memory_format`](https://github.com/pytorch/pytorch/blob/854657f330c46fb698615946874714cfca0ea5dc/torch/_functorch/_aot_autograd/runtime_wrappers.py#L2501)

Because an expanded tensor is not considered contiguous it will be materialized which leads to a difference compared to the eager path.
Example failure caused by this in `TestEagerFusionOpInfoCPU.test_aot_autograd_symbolic_exhaustive_linalg_householder_product_cpu_float32`:

> Mismatched elements: 2 / 30 (6.7%)
> Greatest absolute difference: 5.340576171875e-05 at index (5, 3) (up to 1e-05 allowed)
> Greatest relative difference: 5.707081527361879e-06 at index (5, 2) (up to 1.3e-06 allowed)

But before addressing this I'd like to properly document `is_contiguous` in a way that covers expanded tensors with all or some strides being zero. One could say that for the case of all-zero strides the tensor is still contiguous: There is no gap between elements.
Contrary a row-expanded tensor is not contiguous as after iterating over a row you'd need to jump back.

### Suggest a potential alternative/fix

_No response_

cc @svekars @sekyondaMeta @AlannaBurke @chauhang @penguinwu

### PR DESCRIPTION
**Pull request by dwan-ith**
Fixes #190038.

Updates Tensor.is_contiguous() documentation to precisely define contiguity, explain why zero-stride expanded tensors are not contiguous, and includes a verified example.

### CODE REVIEW

File: torch/_tensor_docs.py [hunk 1]
```diff
@@ -2704,8 +2704,19 @@ def add_docstr_all(method: str, docstr: str) -> None:
     r"""
 is_contiguous(memory_format=torch.contiguous_format) -> bool
 
-Returns True if :attr:`self` tensor is contiguous in memory in the order specified
-by memory format.
+Returns ``True`` if :attr:`self` tensor's elements are stored in a non-overlapping,
+dense layout in the order specified by the memory format. For the default
+``torch.contiguous_format``, this is the standard row-major (C-contiguous) order.
```
Comment (jbschlosser) [inline torch/_tensor_docs.py:2709]: sorry what does "C-contiguous" mean?

Comment (dwan-ith) [reply torch/_tensor_docs.py:2709]: the usual row-major memory layout used by C? (elements are stored so that the last dimension changes fastest in memory)
